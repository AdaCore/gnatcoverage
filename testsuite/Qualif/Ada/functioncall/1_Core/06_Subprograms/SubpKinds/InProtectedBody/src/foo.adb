pragma Ada_2012;

with Ada.Text_IO; use Ada.Text_IO;

package body Foo is

   protected body Obj_Type is

      function F return Integer is                       -- # fun
      begin
         return 8;                                       -- # stmt
      end F;

      function E (I : Integer) return Boolean is (True); -- # expr_fun

      procedure P (S : String) is                        -- # fun
      begin
         Put_Line (S);                                   -- # call_stmt
         null;                                           -- # stmt
      end P;

      procedure NP is                                     -- # fun
         null;                                           -- # stmt
   end Obj_Type;

end Foo;
