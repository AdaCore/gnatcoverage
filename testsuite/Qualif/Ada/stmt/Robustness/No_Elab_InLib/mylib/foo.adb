package body Foo is
   procedure Set_X (V : Myint) is
   begin
      X := V; -- # body
   end;
end;
