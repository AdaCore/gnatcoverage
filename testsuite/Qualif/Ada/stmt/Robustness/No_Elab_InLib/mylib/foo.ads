pragma Restrictions (No_Elaboration_Code);

package Foo is
   type Myint is new Integer;   -- # decl
   X : Myint := 12;             -- # decl
   procedure Set_X (V : Myint); -- # decl
end;

