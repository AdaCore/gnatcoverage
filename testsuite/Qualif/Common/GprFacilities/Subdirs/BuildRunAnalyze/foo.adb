procedure Foo is
   I : Integer := 0;
   pragma Volatile (I);
begin
   if I > 0 then
      I := 1;
   end if;
end Foo;
