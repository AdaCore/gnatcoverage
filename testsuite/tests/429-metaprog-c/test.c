#define DECL(x) int x = 0;

int
foo ()
{
  DECL (a_main)
#include "foo.h"
}

int
main ()
{
  return 0;
}
