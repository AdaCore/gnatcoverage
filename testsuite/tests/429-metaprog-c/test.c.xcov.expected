33% of 3 lines covered
20% statement coverage (1 out of 5)

Coverage level: stmt
   1 .: #define DECL(x) int x = 0;
   2 .: 
   3 .: int
   4 .: foo ()
   5 .: {
   6 -:   DECL (a_main)
statement "int a_mai..." at 6:3 not executed
  note: in definition of macro DECL at location test.c:1:17
  note: from expansion of macro DECL at location test.c:6:3
   7 -: #include "foo.h"
statement "int a_bar..." at 7:10 not executed
  note: in file bar.h at location bar.h:2:3
  note: from inclusion of bar.h at location foo.h:1:10
  note: from inclusion of foo.h at location test.c:7:10
  note: in definition of macro DECL at location test.c:1:17
  note: from expansion of macro DECL at location bar.h:2:3
statement "int a_foo..." at 7:10 not executed
  note: in file foo.h at location foo.h:2:1
  note: from inclusion of foo.h at location test.c:7:10
  note: in definition of macro DECL at location test.c:1:17
  note: from expansion of macro DECL at location foo.h:2:1
statement "return 0;" at 7:10 not executed
  note: in file foo.h at location foo.h:3:1
  note: from inclusion of foo.h at location test.c:7:10
   8 .: }
   9 .: 
  10 .: int
  11 .: main ()
  12 .: {
  13 +:   return 0;
  14 .: }
